### PR TITLE
Improve language scripts

### PR DESCRIPTION
**Pull request by 3d-gussner**
- [x] Add syntax checks for translations in `lang-check.py`
- [x] Check syntax english origin
- [x] Display correct issue line
- [x] Update `lang-ex/import.sh' to replace chars with diacritics
   - [x] German
   - [x] Czech
   - [x] Hungarian
- [x] Update German translation to use `äöüß`
    - [x] `lang/Firmware_de.po`
    - [x] `lang/new/de.po`
    - [x] `lang/lang_en_de.txt`
- [x] Update documentation in scripts

PFW-1283

### CODE REVIEW

**Comment by 3d-gussner** [comment]
@wavexx Could you please review the changes I made in `lang-check.py`?

**Comment by 3d-gussner** [comment]
@AtillaSVK Can you please review/comment/approve? https://github.com/prusa3d/Prusa-Firmware/pull/3342/files#diff-a24289c9ad15eb88af1103e3112ffcd9dca5beb53390bd094ddac06fa1e792f3R238-R275

File: .gitignore [hunk 1]
```diff
@@ -55,3 +55,20 @@ Firmware/Doc
...
+.DS_Store
+**/.DS_Store
+Firmware/.DS_Store
+Firmware/variant/.DS_Store
+lang/.DS_Store
+lang/po/.DS_Store
+lang/po/new/.DS_Store
+Tests/.DS_Store
+tools/.DS_Store
+tools/lib/.DS_Store
```
Comment (DRracer) [inline .gitignore]: ideally add a newline at the end of this file

File: Firmware/messages.c [hunk 2]
```diff
@@ -111,7 +111,7 @@ const char MSG_STOP_PRINT[] PROGMEM_I1 = ISTR("Stop print"); ////c=18
 const char MSG_STOPPED[] PROGMEM_I1 = ISTR("STOPPED."); ////c=20
 const char MSG_TEMP_CALIBRATION[] PROGMEM_I1 = ISTR("Temp. cal."); ////c=14
 const char MSG_TEMP_CALIBRATION_DONE[] PROGMEM_I1 = ISTR("Temperature calibration is finished and active. Temp. calibration can be disabled in menu Settings->Temp. cal."); ////c=20 r=12
-const char MSG_UNLOAD_FILAMENT[] PROGMEM_I1 = ISTR("Unload filament"); ////Number 1 to 5 is added behind text e.g. "Unload filament" c=16
+const char MSG_UNLOAD_FILAMENT[] PROGMEM_I1 = ISTR("Unload filament"); ////c=18
```
Comment (DRracer) [inline Firmware/messages.c:114]: :thinking: are you sure the original comment is no longer valid - i.e. the text is not used for the MMU as well?
In such a case the `c=16` would make sense

Comment (DRracer) [reply Firmware/messages.c:114]: Oh, it has been used for the SNMM and now it is "guarded" with a compile error (as the SNMM has been broken for a long time)

File: lang/lang-import.sh [hunk 3]
```diff
@@ -99,146 +182,176 @@ fi
...
+ #replace 'é' with 'e'
  sed -i 's/\xc3\xa9/e/g' $LNG'_filtered.po'
 fi
 #replace in polish translation
 #if [ "$LNG" = "pl" ]; then
 #fi
 
-#check for nonasci characters
-if grep --color='auto' -P -n '[^\x00-\x7F]' $LNG'_filtered.po' >nonasci.txt; then
+#check for nonasci characters excpet HD44780 ROM A00 'äöüß'
```
Comment (DRracer) [inline lang/lang-import.sh]: /nitpick/: typo `excpet` -> `except`

File: lang/progmem.sh [hunk 4]
```diff
@@ -111,6 +121,6 @@ cat $PROGMEM.chr | \
 #this step can be omitted because .txt file is not used
 cat $PROGMEM.var | sed 's/\r/\n/g' | sed -E 's/^[0-9a-f]{8} [^ ]* //' >$PROGMEM.txt
 
-echo "progmem.sh finished" >&2
+echo "$(tput setaf 2)progmem.sh finished$(tput sgr0)" >&2
 
 exit 0
```
Comment (DRracer) [inline lang/progmem.sh]: no newline at end of file

File: lang/textaddr.sh [hunk 5]
```diff
@@ -63,6 +73,6 @@ text=''
  fi
 done > textaddr.txt
 
-echo "textaddr.sh finished" >&2
+echo "$(tput setaf 2)textaddr.sh finished$(tput sgr0)" >&2
 
 exit 0
```
Comment (DRracer) [inline lang/textaddr.sh]: no newline at end of file

**Review by DRracer — CHANGES_REQUESTED** [review]
just a few minor comments, please have a look at them

**Review by DRracer — APPROVED** [review]
:+1:

**Pull request merged**